packages/react-cosmos-playground/src/components/ResponsiveLoader/index.js

```diff
@@ -0,0 +1,183 @@
+// @flow
+
+import React from 'react';
+import Header from './Header';
+import classNames from 'classnames';
+import localForage from 'localforage';
+import styles from './index.less';
+
+type Props = {
+  inputRef: (node: ?HTMLElement) => void,
+  src: string,
+  showResponsiveControls: boolean,
+  onFixtureUpdate: any,
+  devices: Array<{| label: string, width: number, height: number |}>,
+  fixture: Object
+};
+
+type State = {
+  containerWidth: number,
+  containerHeight: number,
+  savedWidth: ?number,
+  savedHeight: ?number
+};
+
+const RESPONSIVE_FIXTURE_WIDTH = '__cosmos__responsive-fixture-width';
+const RESPONSIVE_FIXTURE_HEIGHT = '__cosmos__respinsive-fixture-height';
+
+const PADDING = 16;
+const BORDER_WIDTH = 2;
+
+class ResponsiveLoader extends React.Component<Props, State> {
+  state = {
+    containerWidth: 10000,
+    containerHeight: 10000,
+    savedWidth: null,
+    savedHeight: null
+  };
+  scalableDiv: ?HTMLElement;
+
+  async componentDidMount() {
+    window.addEventListener('resize', this.updateContainerWidth);
+    setTimeout(this.updateContainerWidth, 1000);
```

Comment:
Why is this scheduled one second in the future?

Reply:
Ah, maybe you have a good solution for this. When `componentDidMount` fires, `this.scalableDiv.getBoundingClientRect()` returns a 0x0 `DOMRect`. But if we wait 1 second, it then appears to have a dimension. I'm not entirely sure what's going on, any ideas would be useful.

scalableDiv is the div that takes up the entirety of the usable space underneath the responsive header all the way to the bottom of the window. We use the size of this to determine how much we need to scale the iFrame wrapper by to fit into this scalableDiv all at once.

Reply:
Did you try calling it in the callback ref of the associated div? You're already fetching the div ref so you could turn it into:

```js
ref={el => {
  this.scalableDiv = el;
  if (el) {
    this.updateContainerWidth()
  }
}}
```